dota2/features/party.py

```diff
@@ -0,0 +1,216 @@
+
+        jobid = self.send_job(EDOTAGCMsg.EMsgClientToGCSetPartyLeader, {
+            "new_leader_steamid": steam_id
+        })
+
+        @self.once(jobid)
+        def wrap_set_party_leader(message):
+            self.emit('set_party_leader', steam_id, message)
+
+        return jobid
+
+    def set_party_coach(self, coach=False):
+        """
+        Set the bot's status as a coach.
+
+        :param coach: bool
+        :return: job event id
+        :rtype: str
+
+        Response event: ``party_coach``
+
+        :param steam_id: steam_id for response
+        :type steam_id: :class:`int`
+
+        :param message: `CMsgDOTAPartyMemberSetCoach <https://github.com/ValvePython/dota2/blob/master/protobufs/dota_gcmessages_client_match_management.proto#L341>`_ proto message
+        """
+        if not self.party:
+            if self.verbose_debug:
+                self._LOG.debug("set_party_coach called when not in a party!")
+            return False
+
+        if self.verbose_debug:
+            self._LOG.debug("Setting coach slot: %s" % coach)
+
+        jobid = self.send_job(EDOTAGCMsg.EMsgGCPartyMemberSetCoach, {
+            "wants_coach": coach
+        })
+
+        @self.once(jobid)
+        def wrap_party_coach(message):
+            self.emit('party_coach', coach, message)
+
+        return jobid
+
+    def invite_to_party(self, steam_id=None):
+        """
+        Invites a player to a party. This will create a new party
+        if you aren't in one.
+
+        :param steam_id: steam_id
+        :return: job event id
+        :rtype: str
+
+        Response event: ``invite_to_party``
+
+        :param steam_id: steam_id for response
+        :type steam_id: :class:`int`
+
+        :param message: `CMsgInviteToParty <https://github.com/ValvePython/dota2/blob/master/protobufs/base_gcmessages.proto#L80>`_ proto message
+        """
+        if not steam_id:
+            if self.verbose_debug:
+                self._LOG.debug("Steam ID required to create a party invite.")
+            return False
+
+        if self.verbose_debug:
+            self._LOG.debug("Inviting %s to a party." % steam_id)
+
+        jobid = self.send_job(EGCBaseMsg.EMsgGCInviteToParty, {
+            "steam_id": steam_id
+        })
+
+        @self.once(jobid)
+        def wrap_invite_to_party(message):
+            self.emit('invite_to_party', steam_id, message)
+
+        return jobid
+
+    def kick_from_party(self, steam_id=None):
+        """
+        Kicks a player from the party. This will create a new party
+        if you aren't in one.
+
+        :param steam_id: steam_id
+        :return: job event id
+        :rtype: str
+
+        Response event: ``kick_from_party``
+
+        :param steam_id: steam_id for response
+        :type steam_id: :class:`int`
+
+        :param message: `CMsgKickFromParty <https://github.com/ValvePython/dota2/blob/master/protobufs/base_gcmessages.proto#L114>`_ proto message
+        """
+        if not steam_id:
```

Comment:
Not having a default value removes the need for this check. There is no need to have a debug message here.